clients/manifest.go

```diff
@@ -33,27 +37,78 @@ func DownloadRetryBackoffLong() backoff.BackOff {
 var DownloadRetryBackoff = DownloadRetryBackoffLong
 
 func DownloadRenditionManifest(requestID, sourceManifestOSURL string) (m3u8.MediaPlaylist, error) {
-	var playlist m3u8.Playlist
-	var playlistType m3u8.ListType
+	playlist, playlistType, _, err := downloadManifest(requestID, sourceManifestOSURL)
+	if err != nil {
+		return m3u8.MediaPlaylist{}, err
+	}
+	return convertToMediaPlaylist(playlist, playlistType)
+}
+
+func DownloadRenditionManifestWithBackup(requestID string, primaryManifestURL, osTransferURL *url.URL) (*url.URL, error) {
+	manifestURL, playlist, playlistType, err := downloadManifestWithBackup(requestID, primaryManifestURL.String())
+	if err != nil {
+		return nil, fmt.Errorf("error downloading manifest: %w", err)
+	}
+	mediaPlaylist, err := convertToMediaPlaylist(playlist, playlistType)
+	if err != nil {
+		return nil, err
+	}
+
+	// If we had to fall back to the backup manifest then we need a new manifest with absolute URLs because
+	// successful segment uploads will exist in the primary store and therefore the relative URLs will be broken.
+	newPlaylistNeeded := manifestURL != primaryManifestURL.String()
 
+	// Check whether segments are available from primary or backup storage
 	dStorage := NewDStorageDownload()
-	err := backoff.Retry(func() error {
-		rc, err := GetFile(context.Background(), requestID, sourceManifestOSURL, dStorage)
+	for _, segment := range mediaPlaylist.GetAllSegments() {
+		segURL, err := ManifestURLToSegmentURL(primaryManifestURL.String(), segment.URI)
```

Comment:
It needs to be the primary because the `GetFileWithBackup` relies on the url being a primary url, to be able to find the backup location for it. So we always check the segment exists on primary first then check backup if it's not there. Does that make sense?

Reply:
Makes sense!